ktor-client/ktor-client-core/common/src/io/ktor/client/request/HttpRequest.kt

```diff
@@ -143,6 +164,21 @@ class HttpRequestData internal constructor(
     val executionContext: Job,
     val attributes: Attributes
 ) {
+    /**
+     * Retrieve extension by it's type.
+     */
+    @KtorExperimentalAPI
+    fun <T> getCapabilityOrNull(key: HttpClientEngineCapability<T>): T? {
+        @Suppress("UNCHECKED_CAST")
+        return attributes.getOrNull(ENGINE_CAPABILITIES_KEY)?.get(key) as T?
+    }
+
+    /**
+     * Retrieve all extension keys associated with this request.
+     */
+    internal fun getRequiredCapabilities(): Set<HttpClientEngineCapability<*>> =
```

Comment:
Not sure, should it be a function or a property?

Reply:
Good point, I made it a property.